akka-http-core/src/main/scala/akka/http/impl/engine/http2/Http2ServerDemux.scala

```diff
@@ -245,7 +245,17 @@ private[http2] class Http2ServerDemux(http2Settings: Http2CommonSettings, initia
           case Setting(Http2Protocol.SettingIdentifier.SETTINGS_MAX_FRAME_SIZE, value) =>
             multiplexer.updateMaxFrameSize(value)
           case Setting(Http2Protocol.SettingIdentifier.SETTINGS_MAX_CONCURRENT_STREAMS, value) =>
+            // TODO: respect the peer settings
             debug(s"Setting max concurrent streams to $value (not enforced)")
+          case Setting(Http2Protocol.SettingIdentifier.SETTINGS_ENABLE_PUSH, value) =>
+            // TODO: respect the peer settings
+            value match {
+              case 0 => debug(s"Client disabled PUSH (ignored, PUSH not supported)")
+              case 1 => debug(s"Client enabled PUSH (ignored, PUSH not supported)")
+              case _ =>
+                pushGOAWAY(PROTOCOL_ERROR, s"Invalid value for SETTINGS_ENABLE_PUSH: $value")
+                settingsAppliedOk = false
```

Comment:
This will introduce a change in behavior. The server is not a bit more strict.

If client sends valid values for the setting they won't notice.

Reply:
Hmm, I implemented this with a server point of view, but when this code runs on the client, the behavior should differ. I need to improve this...